api/src/main/java/edu/wpi/first/shuffleboard/api/util/NetworkTableUtils.java

```diff
@@ -148,13 +148,16 @@ public static DataType dataTypeForEntry(String key) {
 
   /**
    * Waits for ntcore listeners to be fired. This is a <i>blocking operation</i>.
+   *
+   * @param runnable A single NetworkTable call to wait on.
    */
-  public static void waitForNtcoreEvents() {
+  public static void waitForNtcoreEvents(Runnable runnable) {
     CompletableFuture<?> future = new CompletableFuture<>();
     NetworkTablesJNI.addEntryListener("", (uid, key, value, flags) -> {
       future.complete(null);
       NetworkTablesJNI.removeEntryListener(uid);
     }, 0xFF);
+    runnable.run();
```

Comment:
> This problem seems intractable to me if we're using a live networktable.

You could figure it out but you would need to compare the key and value and make sure that the event matches the change.